qiskit/algorithms/time_evolvers/trotterization/trotter_qrte.py

```diff
@@ -0,0 +1,166 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An algorithm to implement a Trotterization real time-evolution."""
+
+from __future__ import annotations
+
+import warnings
+
+from qiskit.algorithms.observables_evaluator import estimate_observables
+
+from qiskit import QuantumCircuit
+from qiskit.algorithms.time_evolvers.time_evolution_problem import TimeEvolutionProblem
+from qiskit.algorithms.time_evolvers.time_evolution_result import TimeEvolutionResult
+from qiskit.algorithms.time_evolvers.real_time_evolver import RealTimeEvolver
+from qiskit.opflow import PauliSumOp
+from qiskit.circuit.library import PauliEvolutionGate
+from qiskit.primitives import BaseEstimator
+from qiskit.quantum_info import Pauli
+from qiskit.synthesis import ProductFormula, LieTrotter
+
+
+class TrotterQRTE(RealTimeEvolver):
+    """Quantum Real Time Evolution using Trotterization.
+    Type of Trotterization is defined by a ``ProductFormula`` provided.
+
+    Attributes:
+        product_formula: A Lie-Trotter-Suzuki product formula. The default is the Lie-Trotter
+            first order product formula with a single repetition.
+
+    Examples:
+
+        .. code-block:: python
+
+            from qiskit.opflow import PauliSumOp
+            from qiskit.quantum_info import Pauli, SparsePauliOp
+            from qiskit import QuantumCircuit
+            from qiskit.algorithms.time_evolvers import TimeEvolutionProblem, TrotterQRTE
```

Comment:
I would argue the TimeEvolutionProblem import should be `from qiskit.algorithms import TimeEvolutionProblem` since we import it at and the API ref documents at that level. (TrotterQRTE depends on the comment in the above init file and it may stay this way if that is the solution chosen)